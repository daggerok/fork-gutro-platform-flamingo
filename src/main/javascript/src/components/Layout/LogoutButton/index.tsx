import React, { useContext } from 'react';
import { Button } from 'antd';

import { AuthenticationContext } from '~/components/Authentication/AuthenticationContextProvider';
import { logoutPlayer } from '~/api/auth';

const LogoutButton: React.FC = () => {
  const { setAuthenticated } = useContext(AuthenticationContext);

  const handleLogoutClick = (): void => {
    setAuthenticated(false);
    logoutPlayer();
  };

  return (
    <Button 
      key="logout-button"
      data-testid="logout-button-wrapper"
      onClick={handleLogoutClick}
      ghost
    >
      Log Out
    </Button>
  );
};

export default LogoutButton;
