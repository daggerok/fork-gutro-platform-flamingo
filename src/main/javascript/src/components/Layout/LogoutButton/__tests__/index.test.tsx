import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

import LogoutButton from '..';

let component: any = null;

jest.mock('./../../../Common/EnvironmentFlag', () => 'environment-flag');

describe('<LogoutButton />', () => {

  it('matches snapshot', () => {

    act(() => {
      component = mount(<LogoutButton />);
    });
    
    expect(component.html()).toMatchSnapshot();
  });

});