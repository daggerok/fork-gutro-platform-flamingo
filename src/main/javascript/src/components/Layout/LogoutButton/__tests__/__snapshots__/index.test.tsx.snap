// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoutButton /> matches snapshot 1`] = `"<button data-testid=\\"logout-button-wrapper\\" type=\\"button\\" class=\\"ant-btn ant-btn-background-ghost\\"><span>Log Out</span></button>"`;
